packages/adminrouter/extra/src/README.md

```diff
@@ -261,31 +261,192 @@ as-is/usually do not follow any of these patterns. It is possible to standardise
 it, but it does not seem to be justified. For example:
  * `util.lua` directly exposes its functions as they are stateless and shared
    between both flavours
- * `lib/auth/open.lua` uses the `.init()` pattern in order to achive different
+ * `lib/auth/open.lua` uses the `.init()` pattern in order to achieve different
    behaviour depending on the flavour.
 
-## Service Endpoints
-
-Admin Router allows Marathon tasks to define custom service UI and HTTP endpoints, which are made available via `<dcos-cluster>/service/<service-name>`. This can be achieved by setting the following Marathon task labels:
-
+## Service Endpoint
+Admin Router offers the `/service` endpoint which enables users to easily
+access some of the tasks launched on DC/OS. AR internally pre-fetches data from
+Mesos and root Marathon and stores it in the internal cache for later use when
+routing an incoming request.
+
+Marathon tasks can be accessed by application ID, Mesos frameworks can be
+accessed by application ID or by framework ID. For example - let's deploy a
+SchmetterlingDB framework with application ID of
+`/group1/group2/schmetterlingA`. It will be accessible using its framework ID
+(let's assume that Mesos assigned it a framework ID of
+`819aed93-4143-4291-8ced-5afb5c726803-0000`):
 ```
-"labels": {
-    "DCOS_SERVICE_NAME": "service-name",
-    "DCOS_SERVICE_PORT_INDEX": "0",
-    "DCOS_SERVICE_SCHEME": "http"
-  }
+/service/819aed93-4143-4291-8ced-5afb5c726803-0000/
+```
+and using its application ID:
+```
+/service/group1/group2/schmetterlingA/
 ```
 
-When your container/task has its own IP (typically when running in a virtual network), `http://<dcos-cluster>/service/service-name` would be forwarded to the container/task's IP using one of the ports in the port mapping or port definition when USER networking is enabled or one of the discovery ports otherwise. When your task/container is mapped to the host, it would be forwarded to the host running the task using the one of the ports allocated to the task. The chosen port is defined by the DCOS_SERVICE_PORT_INDEX label.
-
-In order for the forwarding to work reliably across task failures, we recommend co-locating the endpoints with the task. This way, if the task is restarted on a potentially other host and with different ports, Admin Router will pick up the new labels and update the routing. NOTE: Due to caching there might be an up to 30-second delay until the new routing is working.
-
-We would recommend having only a single task setting these labels for a given `service-name`.
-In the case of multiple task instances with the same `service-name` label, Admin Router will pick one of the tasks instances deterministically, but this might make debugging issues more difficult.
+The `/service` endpoint supports WebSockets. By default, it disables NGINX
+request and response buffering. All redirects from services are rewritten in
```

Comment:
If we really disable request buffering on this service in general I'd like to understand why -- asked Pawel in person.